polaris-react/src/components/ActionMenu/components/MenuGroup/MenuGroup.tsx

```diff
@@ -42,6 +42,14 @@ export function MenuGroup({
     onClose(title);
   }, [onClose, title]);
 
+  const handleCloseActionList = useCallback(
```

Comment:
@knames  I tried setting it up on my spin instance with the Discounts `Export` button in the `DiscountList`.

The component already has a `ref` created and passed to the `ExportModal` as an `activator`, the only thing I added was passing the `ref` and `id` to `secondaryActions` and [publishing your polaris changes to web with yalc](https://web.docs.shopify.io/docs/guides/working-with-polaris). 
- [Spin](https://shop1.shopify.modal-focus-polaris.tatiana-uspenskaia.us.spin.dev/admin/discounts)
- [Consumer changes in web](https://github.com/Shopify/web/commit/d5005b4cb6c4239d89e505b64a9347c1c865d307)

The focus works for the standalone `Export` button but it doesn't work when it's collapsed on resize as it just closes the popover altogether 🤔  Do you know what's up?

https://user-images.githubusercontent.com/15054990/227380979-ca153633-a92a-4bec-87c3-6b6229eb68c4.mp4

Reply:
I'm still unable to break my instance, also I don't have any issues with focus being stuck on the popover, it always goes to the modal and then back to `export` again.

In oksana's work though, i notice focus being stuck on `duplicate` and it flat out not working on the fake app...

Reply:
We've found the issue!

I was testing using tab, esc, and spacebar. As soon as you guys clicked on the X in the modal, the popover was closed. If you just hit esc, the popover isn't closed. The issue is the popover closes as soon as you click anywhere on the screen, even if a modal isn't open. We need to keep the popover open regardless of clicks when a modal is open.

Reply:
chloe with the bomb recomendation of using `preventCloseOnChildOverlayClick`